### ISSUE TITLE
ImportError: cannot import name 'line_optim' from 'line_refinement' (unknown location)

### ISSUE THREAD

**Issue by 2jjjd** [cvg/DeepLSD#28]
there is no such a file named 'line_optim' in the file'line_refinement',
how to  import line_optim from line_refinement in the file        deeplsd/models/deeplsd.py?

**Comment by rpautrat** [comment]
Hi, did you correctly installed deeplsd with the full install, as described in the ReadMe?

**Comment by ogencoglu** [comment]
Hi-jacking this. Does line refinement mean e.g. double lines will be merged into one with the `'optimize': True` config or something else?

**Comment by rpautrat** [comment]
No, the line refinement is only meant to slightly move a bit the lines to make them perfectly fit the predicted distance and angle fields. So you would keep the same number of lines, including double lines.

**Comment by houyiliang** [comment]
Hello, it also appears after I complete the configuration according to full install.
`ImportError: cannot import name 'line_optim' from 'line_refinement' (unknown location)`
how did you solve it? thank you.

**Comment by rpautrat** [comment]
Hi, can you send here the full output that you get after running the full install command?

**Comment by reai000** [comment]
> Hi, can you send here the full output that you get after running the full install command?

![0](https://github.com/cvg/DeepLSD/assets/125547591/3db2bff9-e158-4e9b-94fd-20598db61a61)

Hi, I also encountered the same problem. Where is the definition of line_optim?    Thank you

**Comment by rpautrat** [comment]
line_optim is defined in the `line_refinement` folder, with Python bindings. It is installed during the full install described in the Readme.

Note that you only need it to refine lines, but if you just want to use the base version of DeepLSD for inference, you don't need it. The Readme explains how to do this quickstart install.

**Comment by reai000** [comment]
> line_optim is defined in the `line_refinement` folder, with Python bindings. It is installed during the full install described in the Readme.
> 
> Note that you only need it to refine lines, but if you just want to use the base version of DeepLSD for inference, you don't need it. The Readme explains how to do this quickstart install.
I solved it, thank you very much

**Comment by reai000** [comment]
I solved it, thank you very much。

**Comment by whw9483** [comment]
> I solved it, thank you very much。

How was it resolved?
